contracts/ContributionAccountingToken.sol

```diff
@@ -46,31 +53,29 @@ contract ContributionAccountingToken is ERC20Burnable, ERC20Permit, AccessContro
     function maxMintableAmount() public view returns (uint256) {
         uint256 currentSupply = totalSupply();
         
-        // If current supply is less than threshold, return remaining amount to threshold
         if (currentSupply < thresholdSupply) {
             return thresholdSupply - currentSupply;
         }
-        
-        // Calculate based on expansion rate
         uint256 elapsedTime = block.timestamp - lastMintTimestamp;
         uint256 maxMintableAmount = (currentSupply * maxExpansionRate * elapsedTime) / (365 days * 100);
-        
-        // Also check against remaining max supply
         uint256 remainingSupply = maxSupply - currentSupply;
         
         return maxMintableAmount < remainingSupply ? maxMintableAmount : remainingSupply;
     }
 
-    function mint(address to, uint256 amount) public onlyRole(MINTER_ROLE) {
-        
-        // Minting fee calculation
+    function userAmountAfterFees(uint256 amount) public pure returns (uint256) {
```

Comment:
This function is being used only once and it is a very small function. Therefore, it is better to remove this function and just move lines 67 and 68 directly there where they are needed (around line 74).